### PR TITLE
Implements the handler for get-proof-by-hash in CT frontend

### LINKED ISSUE #94: Implement handler for get-proof-by-hash
In the example CT frontend including all the RFC 6962 gubbins.

### PR DESCRIPTION
**Pull request by Martin2112**
Includes tests. Closes #94.

### CODE REVIEW

File: examples/ct/ct_handlers.go [hunk 1]
```diff
@@ -321,13 +330,68 @@ func wrappedGetSTHConsistencyHandler(rpcClient trillian.TrillianLogClient) http.
...
+		ctx, _ := context.WithDeadline(context.Background(), getRPCDeadlineTime(c))
+		response, err := c.rpcClient.GetInclusionProofByHash(ctx, &rpcRequest)
+
+		if err != nil || !rpcStatusOK(response.GetStatus()) {
+			sendHttpError(w, http.StatusInternalServerError, fmt.Errorf("RPC failed, possible extra info: %v", err))
+			return
+		}
+
+		// Per RFC 6962 section 4.5 we can't handle a case with duplicate leaves because the API is
+		// defined to return a single proof so it's an error if we don't get exactly one proof
```
Comment (AlCutter) [inline examples/ct/ct_handlers.go]: In the 6962 domain the right thing to do would be to return the proof to the earlier entry.

**Comment by Martin2112** [comment]
The tree_size param wasn't being handled. That should be fixed now.

File: examples/ct/ct_handlers.go (hunk 1 above)
Comment (Martin2112) [inline examples/ct/ct_handlers.go]: I'm also concerned that we could get a large number of results from storage in weird edge cases.

Comment (Martin2112) [inline examples/ct/ct_handlers.go]: I've clarified the comment and added a TODO to make it do the right thing. Will probably tweak the api for this when I've caught up with everything else.

**Pull request merged**